jsk_perception/node_scripts/split_image.py

```diff
@@ -41,7 +55,7 @@ def _split_cb(self, msg):
                 split_img = img[int(v*v_pixels):int((v+1)*v_pixels),
                                 int(h*h_pixels):int((h+1)*h_pixels)]
                 pub_msg = self.bridge.cv2_to_imgmsg(split_img, encoding='bgr8')
```

Comment:
did you try this part with `passthrough` too?

Reply:
Yes, but I got the following error...
```
$ roslaunch jsk_perception sample_split_image.launch                                                 
... logging to /home/naoya/.ros/log/d097d916-dfc1-11eb-aeec-b808cf4bc0d8/roslaunch-naoya-ThinkPad-T460s-5656.log
Checking log directory for disk usage. This may take a while.                                        
Press Ctrl-C to interrupt                         
WARNING: disk usage in log directory [/home/naoya/.ros/log] is over 1GB.                             
It's recommended that you use the 'rosclean' command.                                                

started roslaunch server http://192.168.0.21:41585/                                                  

SUMMARY                  
========                 

PARAMETERS               
 * /image_publisher/encoding: rgb8                
 * /image_publisher/file_name: /home/naoya/ros/m...                                                  
 * /image_publisher/fovx: 84.1                    
 * /image_publisher/fovy: 53.8                    
 * /image_publisher/publish_info: True            
 * /rosdistro: melodic   
 * /rosversion: 1.14.9   
 * /split_image/horizontal_parts: 2               
 * /split_image/vertical_parts: 2                 

NODES                    
  /                      
    image_publisher (jsk_perception/image_publisher.py)                                              
    image_view_split_0_0 (image_view/image_view)  
    image_view_split_0_1 (image_view/image_view)  
    image_view_split_1_0 (image_view/image_view)  
    image_view_split_1_1 (image_view/image_view)  
    split_image (jsk_perception/split_image.py)   

auto-starting new master 
process[master]: started with pid [5690]          
ROS_MASTER_URI=http://localhost:11311             

setting /run_id to d097d916-dfc1-11eb-aeec-b808cf4bc0d8                                              
process[rosout-1]: started with pid [5706]        
started core service [/rosout]                    
process[image_publisher-2]: started with pid [5713]                                                  
process[split_image-3]: started with pid [5717]   
process[image_view_split_0_0-4]: started with pid [5718]                                             
process[image_view_split_0_1-5]: started with pid [5719]                                             
process[image_view_split_1_0-6]: started with pid [5720]                                             
process[image_view_split_1_1-7]: started with pid [5721]                                             
[ERROR] [1625730913.131604365]: Unable to convert '8UC3' image for display: 'cv_bridge.cvtColorForDisplay() while trying to convert image from '8UC3' to 'bgr8' an exception was thrown ([8UC3] is not a color format. but [bgr8] is. The conversion does not make sense)'
[ERROR] [1625730913.152261146]: Unable to convert '8UC3' image for display: 'cv_bridge.cvtColorForDisplay() while trying to convert image from '8UC3' to 'bgr8' an exception was thrown ([8UC3] is not a color format. but [bgr8] is. The conversion does not make sense)'
[ERROR] [1625730913.152328407]: Unable to convert '8UC3' image for display: 'cv_bridge.cvtColorForDisplay() while trying to convert image from '8UC3' to 'bgr8' an exception was thrown ([8UC3] is not a color format. but [bgr8] is. The conversion does not make sense)'
[ERROR] [1625730913.164210453]: Unable to convert '8UC3' image for display: 'cv_bridge.cvtColorForDisplay() while trying to convert image from '8UC3' to 'bgr8' an exception was thrown ([8UC3] is not a color format. but [bgr8] is. The conversion does not make sense)'
[ERROR] [1625730943.157661813]: Unable to convert '8UC3' image for display: 'cv_bridge.cvtColorForDisplay() while trying to convert image from '8UC3' to 'bgr8' an exception was thrown ([8UC3] is not a color format. but [bgr8] is. The conversion does not make sense)'
[ERROR] [1625730943.158164121]: Unable to convert '8UC3' image for display: 'cv_bridge.cvtColorForDisplay() while trying to convert image from '8UC3' to 'bgr8' an exception was thrown ([8UC3] is not a color format. but [bgr8] is. The conversion does not make sense)'
[ERROR] [1625730943.168391956]: Unable to convert '8UC3' image for display: 'cv_bridge.cvtColorForDisplay() while trying to convert image from '8UC3' to 'bgr8' an exception was thrown ([8UC3] is not a color format. but [bgr8] is. The conversion does not make sense)'
```

Reply:
ah, i see. 
so if you use `passthrough` in `imgmsg_to_cv2`, you should check encoding from `msg.encoding` and set the same encoding in here.

Reply:
Thank you! I understand.

With this commit, no error occurs.
https://github.com/jsk-ros-pkg/jsk_recognition/pull/2573/commits/5040230da754f3f8e33a95491474f0aff8952593

However, CPU usage is still over 100%...

Reply:
if so we need C++ implementation